### PR TITLE
SecurityContext deprecated from symfony 2.6

### PR DESCRIPTION
**Pull request by danilupion**
Updated security component references to avoid deprecated usage http://symfony.com/blog/new-in-symfony-2-6-security-component-improvements

### CODE REVIEW

**Comment by danilupion** [comment]
I have also updated Twig/SecurityExtension.php so it no longer uses deprecated since Twig 1.2 Twig_Function_Method method.

File: Resources/config/security_expressions.xml [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
     <services>
         <service id="security.extra.twig_extension" class="%security.extra.twig_extension.class%" public="false">
-            <argument type="service" id="security.context" />
+            <argument type="service" id="security.authorization_checker" />
```
Comment (stof) [inline Resources/config/security_expressions.xml]: this means you are dropping support for Symfony < 2.6. you should either change the composer.json to make it part of the requirements, or keep support for older versions in parallel (which is possible and may be better as 2.3 is still a maintained version)

File: Security/Authorization/Interception/MethodSecurityInterceptor.php [hunk 2]
```diff
@@ -50,11 +50,11 @@ class MethodSecurityInterceptor implements MethodInterceptorInterface
     private $runAsManager;
     private $logger;
 
-    public function __construct(SecurityContextInterface $securityContext, AuthenticationManagerInterface $authenticationManager, AccessDecisionManagerInterface $accessDecisionManager,
+    public function __construct(TokenStorage $tokenStorage, AuthenticationManagerInterface $authenticationManager, AccessDecisionManagerInterface $accessDecisionManager,
```
Comment (stof) [inline Security/Authorization/Interception/MethodSecurityInterceptor.php]: typehint the implementation is a total no-go

File: Security/Authorization/Interception/MethodSecurityInterceptor.php [hunk 3]
```diff
@@ -50,11 +50,11 @@ class MethodSecurityInterceptor implements MethodInterceptorInterface
     private $runAsManager;
     private $logger;
 
-    public function __construct(SecurityContextInterface $securityContext, AuthenticationManagerInterface $authenticationManager, AccessDecisionManagerInterface $accessDecisionManager,
+    public function __construct(TokenStorageInterface $tokenStorage, AuthenticationManagerInterface $authenticationManager, AccessDecisionManagerInterface $accessDecisionManager,
```
Comment (danilupion) [inline Security/Authorization/Interception/MethodSecurityInterceptor.php]: I completely agree with your no-go statement, I was a rushed changed, sorry for that.

File: Resources/config/security_expressions.xml (hunk 1 above)
Comment (danilupion) [inline Resources/config/security_expressions.xml]: I am not too experienced in symfony, how would you keep support for older versions? you just mean sticking to the 2.6 deprecated security.context, right?

Comment (stof) [inline Resources/config/security_expressions.xml]: you would have to use a compiler pass to change what gets injected in the service. and your constructor should accept both the old and new classes (meaning no typehint but some instanceof check inside the constructor)

**Comment by acasademont** [comment]
+1 please!

**Comment by tobias-93** [comment]
:+1:

**Comment by XWB** [comment]
Ping @schmittjoh

**Comment by Phobetor** [comment]
:+1:

**Comment by mgeorgiev-rmn** [comment]
+1 please

**Comment by thomask** [comment]
+1

**Comment by gaydarov** [comment]
+1

**Comment by dspiegel** [comment]
Ping @schmittjoh

**Comment by stof** [comment]
should be closed in favor of #198 which finishes it

**Pull request closed without merging**